src/Classes/Item.lua

```diff
@@ -63,6 +63,198 @@ function ItemClass:ResetInfluence()
+										local excluded = false
+										-- if we have an exclusion pattern list for that tagName and itemSlotName
+										if data.itemTagSpecialExclusionPattern[tagName] and data.itemTagSpecialExclusionPattern[tagName][itemSlotName] then
+											-- iterate across the exclusion patterns
+											for _, specialMod in ipairs(data.itemTagSpecialExclusionPattern[tagName][itemSlotName]) do
+												-- and if the description matches pattern exclude it
+												if dv:lower():find(specialMod:lower()) then
+													excluded = true
+													break
+												end
+											end
+										end
+										if excluded then
+											found_2 = true
+											break
+										end
+									end
+								end
+								if not found_2 and not inverseModifierFoundList[k] then
+									inverseModifierFoundList[k] = true
+									ConPrintf("[%s] appears in desc but not in tags. [%s] %s", tag_name, k, dv)
+									break
+								end
+							end
+						end
+					end
+				else
+					-- this is useless if the modTags = { } (is empty)
+					if #v.modTags > 0 then
+						for _,tag in ipairs(v.modTags) do
+							if tag:lower() == tag_name then
+								found_2 = true
+								break
+							else
+								local excluded = false
+								-- if we have an exclusion pattern list for that tagName and itemSlotName
+								if data.itemTagSpecialExclusionPattern[tagName] and data.itemTagSpecialExclusionPattern[tagName][itemSlotName] then
+									-- iterate across the exclusion patterns
+									for _, specialMod in ipairs(data.itemTagSpecialExclusionPattern[tagName][itemSlotName]) do
+										-- and if the description matches pattern exclude it
+										if dv:lower():find(specialMod:lower()) then
+											excluded = true
+											break
+										end
+									end
+								end
+								if excluded then
+									found_2 = true
+									break
+								end
+							end
+						end
+						if not found_2 and not inverseModifierFoundList[k] then
+							inverseModifierFoundList[k] = true
+							ConPrintf("[%s] appears in desc but not in tags. [%s] %s", tag_name, k, dv)
+						end
+					end
+				end
+			end
+		end
+	end
+end
+
+-- Iterate over modifiers to see if specific substring is found (for conditional checking)
+function ItemClass:FindModifierSubstring(substring, itemSlotName)
+	local modLines = {}
+
+	-- The commented out line below is used at GGPK updates to check if any new modifiers
+	-- have been identified that need to be added to the manually maintained special modifier
+	-- pool in Data.lua (data.itemTagSpecial and data.itemTagSpecialExclusionPattern tables)
+	--getTagBasedModifiers(substring, itemSlotName)
+
+	-- merge various modifier lines into one table
+	for k,v in pairs(self.enchantModLines) do modLines[k] = v end
+	for k,v in pairs(self.scourgeModLines) do modLines[k] = v end
+	for k,v in pairs(self.implicitModLines) do modLines[k] = v end
+	for k,v in pairs(self.explicitModLines) do modLines[k] = v end
```

Comment:
Hmm, do crucible mods belong here as well?  We'll probably want to test how that works.  You're also missing crafted mods in this block.